packages/simcore-sdk/src/simcore_sdk/node_ports_common/storage_client.py

```diff
@@ -76,8 +76,26 @@ async def wrapped(*args, **kwargs):
 @lru_cache
 def _base_url() -> str:
     settings = NodePortsSettings.create_from_envs()
-    base_url: str = settings.NODE_PORTS_STORAGE.api_base_url
-    return base_url
+    return (
+        settings.NODE_PORTS_STORAGE_AUTH.base_url
+        if settings.NODE_PORTS_STORAGE_AUTH
+        else settings.NODE_PORTS_STORAGE.api_base_url
+    )
+
+
+@lru_cache
```

Comment:
It makes the trick but I  find it very ugly/hacky to add a cache to "store" a state :-) An lru_cache is intended for some cpu-like operation etc ... while here the idea is to store a state during a lifespan. There is no scoped object to store (e.g. like an app or client object) in simcore-sdk? I wonder whether https://docs.python.org/3/library/contextvars.html is the clean way to go here

Reply:
While reading it, I think that a cache is pretty obvious and readable to the purpose of what it's supposed to do. Previously this `lru_cache` was called `cache` but was deprecated. It would have resulted more readable.

We have no session object in simcore sdk unfortunately, that's why we have to resort to something like this cache.

While contextvars seem interesting, they will add a lot more code to support the exact same result. I still lack some understanding on how they work and such a change should go in a separate PR.

Reply:
its not about "obvious and readable" what it does but about purpose. I still think that the purpose of a cache is something else :-)